vulture/core.py

```diff
@@ -395,6 +395,10 @@ def visit_ClassDef(self, node):
         else:
             self._define(
                 self.defined_classes, node.name, node, ignore=_ignore_class)
+    
+    def visit_Constant(self, node):
+        if getattr(node, 's'):
+            self.visit_Str(node)
```

Comment:
I think we should do `getattr(node, 's', None)` otherwise it will throw an error whenever it won't find `'s'` as an attribute.

Other than that, it looks good to me.

Reply:
`hasattr(node, 's')` would be even better.

But the Constant node never has a `s` attribute. I think the correct way would be:

```
def visit_Constant(self, node):
    if isinstance(node.value, str):
        self._handle_string(node.value)
```

Also, the visit_Str method should only be defined for Python < 3.8. Once we do that, it will be easier to test the new code.

Reply:
> hasattr(node, 's') would be even better.

Oh, of course! 😅

> Also, the visit_Str method should only be defined for Python < 3.8. Once we do that, it will be easier to test the new code.

Hmm, interesting!

I'm not sure how we can conditionally define a function in Python, but I am thinking somewhere along the lines of:

```python
def define_if_has_Str(f):
    if sys.version_info >= (3, 8):
        return f
    del f    # Not sure if it deletes the global reference or just locally
    def _f(*args):
        pass
    return _f


class Vulture:
    # ...

    @define_if_has_Str
    def visit_Str(self, node):
        self._handle_string(node.s)
```

@jendrikseipp is this too hacky?

What all other options do we have otherwise?